application/controllers/UserGroupController.php

```diff
@@ -100,98 +100,96 @@ public function actionIndex()
      */
     public function actionViewGroup($ugid, bool $header = false)
     {
-        if (!Permission::model()->hasGlobalPermission('usergroups', 'read')) {
+        if (empty(Yii::app()->session['loginID']) || !Permission::model()->hasGlobalPermission('usergroups', 'read')) {
             Yii::app()->session['flashmessage'] = gT('Access denied!');
             $this->redirect(App()->createUrl("/admin"));
         }
 
+        $ugid = (int)$ugid;
+        if (empty($ugid)) {
+            Yii::app()->user->setFlash('error', gT('GroupId missing'));
+            $this->redirect('index');
+        }
+
+        $userGroup = UserGroup::model()->findByPk($ugid);
+        if (empty($userGroup)) {
+            throw new CHttpException(404, gT("User group not found."));
+        }
+
+        // Only allow access if user is:
+        // - Superadmin
+        // - Owner of the group
+        // - Member of the group
+        if (
+            !(
+                $userGroup->owner_id == Yii::app()->user->id ||
```

Comment:
No usage of `usercontrolSameGroupPolicy` ? Must review according to https://bugs.limesurvey.org/view.php?id=18289 decision

If we use ownerID : https://github.com/LimeSurvey/LimeSurvey/blob/master/application/models/Interfaces/PermissionInterface.php#L14 


```
        /* If have global : return true */
        if (!App()->getConfig('usercontrolSameGroupPolicy') && Permission::model()->hasPermission(0, 'global', 'usergroups', $sCRUD, $iUserID)) {
            return true;
        }
        /* Finally : return specific one */
        /* To check */
    }
```

Reply:
in 3.X :` if (Permission::model()->hasGlobalPermission('usergroups', 'delete')) { `

Reply:
> No usage of usercontrolSameGroupPolicy 

So far, this is only used in the dropdown list for assign permission to a survey.
Adding it here seems a bit out of scope.
Maybe it should, but let's analyze it separately, as to be able to wrap up the PR

Reply:
Yes